client/powerline.c

```diff
@@ -90,8 +91,18 @@ int main(int argc, char *argv[]) {
 	memset(&server, 0, sizeof(struct sockaddr_un));
 	server.sun_family = AF_UNIX;
 	strncpy(server.sun_path A, address, strlen(address));
+#ifdef __linux__
+	/* We use a socket in the abstract namespace on Linux, and because SUN_LEN uses strlen, it does not work for these. The struct size also fails, so compute manually. */
+	socklen = sizeof(server.sun_family) + strlen(address) A;
+#elif defined SUN_LEN
```

Comment:
My headers (Gentoo linux) do contain SUN_LEN. But this is not going to be ever used because of `#ifdef __linux__` being the first.